utilities/src/serialize/impls.rs

```diff
@@ -398,6 +398,46 @@ impl<T: CanonicalDeserialize> CanonicalDeserialize for Vec<T> {
     }
 }
 
+impl<T: CanonicalDeserialize> CanonicalDeserialize for [T; 32] {
+    #[inline]
+    fn deserialize_with_mode<R: Read>(
+        mut reader: R,
+        compress: Compress,
+        validate: Validate,
+    ) -> Result<Self, SerializationError> {
+        let len = u64::deserialize_with_mode(&mut reader, compress, validate)?;
```

Comment:
The length is statically known, so we can avoid serializing and deserializing it.

Reply:
Absolutely, it's a leftover from when I tried to go for a generic approach, removing it